# Géocodage SITG-labs

## Vérifier json API

In [1]:
from sitg_geocode import inspect_sitg_response

await inspect_sitg_response(adresse="Av. J-D Maillard, 7, Meyrin")

{
  "hits": [
    {
      "addressId": "74281_1530_00007",
      "streetName": "Avenue des Vallées",
      "houseNumber": "7",
      "postalCode": "74200",
      "locality": "Thonon-les-Bains",
      "municipality": "Thonon-les-Bains",
      "type": null,
      "EGID": null,
      "EGRID": null,
      "longitude": 6.485875,
      "latitude": 46.370337,
      "coordinates": {
        "x": 967879.55,
        "y": 6591514.21,
        "crs": "EPSG:2154"
      },
      "administrativeDivision": "Département de la Haute-Savoie",
      "country": "France",
      "indexed_at": "2026-07-08T04:00:15.151852+00:00",
      "score": 56.0,
      "dataSource": {
        "provider": "Base Adresse Nationale (BAN)",
        "metadataUrl": "https://adresse.data.gouv.fr/",
        "dataUrl": "https://adresse.data.gouv.fr/data/ban/adresses/latest/csv/adresses-74.csv.gz",
        "retrievedAt": "2026-07-08T04:00:08.517671+00:00"
      },
      "formatted": null
    }
  ],
  "query": "7 avenue j-d maillard me

## Corriger liste d'adresses

In [2]:
import polars as pl
from polars.dataframe.frame import DataFrame

from sitg_geocode import sitg_geocode_async

# lire le CSV de test
df: DataFrame = pl.read_csv(source="test_adresses.csv")

# Geocodage
result_geocode: DataFrame = await sitg_geocode_async(
    df,
    col_adresse="Rue et N°",
    max_concurrent=10,
    min_score_threshold=95,
)

# Joindre et exporter
df_resultat: DataFrame = df.join(other=result_geocode, on="Rue et N°", how="left")
df_resultat.write_csv(file="resultat.csv")

display(df_resultat.to_pandas())

Geocoding SITG:   0%|          | 0/107 [00:00<?, ?it/s]

2026-07-08 11:32:57.056 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Ch.Gustave Rochette 14' (5 résultat(s) hors de ce canton)


2026-07-08 11:32:57.067 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Rue du Simplon, 15' (2 résultat(s) hors de ce canton)


2026-07-08 11:32:57.095 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Rue Jean-Jacque Rousseau 34' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.118 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Rue Joseph Girard, 12, 14' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.197 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Rue Riont Parc 21' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.207 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'rue du Pré-du-Couvent 1bis' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.219 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Place Cornavin 1D' (2 résultat(s) hors de ce canton)


2026-07-08 11:32:57.282 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour '37, sous Garan' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.313 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Av. Blanc, 5' (2 résultat(s) hors de ce canton)


2026-07-08 11:32:57.348 | WARNING  | sitg_geocode.geocode:_fetch_one:161 - Aucun résultat dans 'Canton de Genève' pour 'Rue de la Fontaine 1' (1 résultat(s) hors de ce canton)


2026-07-08 11:32:57.400 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour '12 rue Jean-Charles Amat' (seuil 95) : meilleure proposition 'Rue Jean-Charles-AMAT 12' avec un score de 75.47


2026-07-08 11:32:57.401 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Ancienne route, 78' (seuil 95) : meilleure proposition 'L'Ancienne-Route 78' avec un score de 78.05


2026-07-08 11:32:57.402 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Av. J-D Maillard, 7' (seuil 95) : meilleure proposition 'Avenue KRIEG 7' avec un score de 59.26


2026-07-08 11:32:57.403 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Avenue de Châtelaine 60-62' (seuil 95) : meilleure proposition 'Avenue de Châtelaine 60' avec un score de 85.0


2026-07-08 11:32:57.403 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Avenue du Grand Saleve 2' (seuil 95) : meilleure proposition 'Avenue du Grand-Salève 2' avec un score de 85.11


2026-07-08 11:32:57.404 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Bvd Jaous Fazy 23' (seuil 95) : meilleure proposition 'Boulevard de la Cluse 23' avec un score de 69.57


2026-07-08 11:32:57.404 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Ch de la Bassonette, 9' (seuil 95) : meilleure proposition 'Chemin de la Bessonnette 9' avec un score de 81.82


2026-07-08 11:32:57.405 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin Etienne Chennaz 10' (seuil 95) : meilleure proposition 'Chemin Etienne-CHENNAZ 10' avec un score de 67.74


2026-07-08 11:32:57.406 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin de la Montagne 51-53' (seuil 95) : meilleure proposition 'Chemin De-LA-MONTAGNE 51' avec un score de 72.0


2026-07-08 11:32:57.406 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin de la Paumière,20' (seuil 95) : meilleure proposition 'Chemin de la Paumière 2' avec un score de 76.92


2026-07-08 11:32:57.406 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin du Pont de l'Ecu 1' (seuil 95) : meilleure proposition 'Chemin du Pont 1' avec un score de 83.87


2026-07-08 11:32:57.407 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin sous Blame, 53' (seuil 95) : meilleure proposition 'Chemin de Sous-Balme 53' avec un score de 66.67


2026-07-08 11:32:57.407 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Passage Jean-Francois Deshusses' (seuil 95) : meilleure proposition 'Passage Jean-François-DESHUSSES 2' avec un score de 63.77


2026-07-08 11:32:57.408 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Pellegrino Rossi 8' (seuil 95) : meilleure proposition 'Rue Pellegrino-ROSSI 8' avec un score de 75.56


2026-07-08 11:32:57.408 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Promenade de l’Europe 67' (seuil 95) : meilleure proposition 'Promenade de l'Europe 67' avec un score de 85.11


2026-07-08 11:32:57.409 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Route de Drize,61' (seuil 95) : meilleure proposition 'Route de Lullier 65' avec un score de 55.56


2026-07-08 11:32:57.409 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Route de la Louvière 18' (seuil 95) : meilleure proposition 'Route de La-Louvière 18' avec un score de 80.95


2026-07-08 11:32:57.409 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Route des Chevaliers de Malte 30' (seuil 95) : meilleure proposition 'Route des Chevaliers-de-Malte 30' avec un score de 82.76


2026-07-08 11:32:57.410 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Route du Roussillon 18' (seuil 95) : meilleure proposition 'Route De-ROSSILLON 18' avec un score de 80.0


2026-07-08 11:32:57.410 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rte de grand Lancy 161' (seuil 95) : meilleure proposition 'Route du Grand-Lancy 161' avec un score de 85.11


2026-07-08 11:32:57.411 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rte du Gd Lancy 166' (seuil 95) : meilleure proposition 'Route de Choulex 166' avec un score de 66.67


2026-07-08 11:32:57.411 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Ami Lévrier, 3' (seuil 95) : meilleure proposition 'Rue Ami-LÉVRIER 3' avec un score de 82.05


2026-07-08 11:32:57.412 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Charles Humbert, 5' (seuil 95) : meilleure proposition 'Rue Charles-HUMBERT 5' avec un score de 85.11


2026-07-08 11:32:57.412 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Dizerens, 7b' (seuil 95) : meilleure proposition 'Rue de la Golette 7B' avec un score de 57.14


2026-07-08 11:32:57.413 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Henry Veyrassat, 7' (seuil 95) : meilleure proposition 'Rue Henri-VEYRASSAT 7' avec un score de 80.85


2026-07-08 11:32:57.413 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Jacques-Grosselin,14' (seuil 95) : meilleure proposition 'Rue Jacques-GROSSELIN 1' avec un score de 82.35


2026-07-08 11:32:57.414 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Jean Charles Amat 22' (seuil 95) : meilleure proposition 'Rue Jean-Charles-AMAT 22' avec un score de 60.38


2026-07-08 11:32:57.414 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Jean Robert Chouet 7' (seuil 95) : meilleure proposition 'Rue Jean-Robert-CHOUET 7' avec un score de 64.15


2026-07-08 11:32:57.415 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Paul Bouchet, 5' (seuil 95) : meilleure proposition 'Rue Paul-BOUCHET 5' avec un score de 63.41


2026-07-08 11:32:57.415 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue de Moillebeau,1' (seuil 95) : meilleure proposition 'Rue de Moillebeau 38' avec un score de 75.68


2026-07-08 11:32:57.415 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue de Zuriche 43' (seuil 95) : meilleure proposition 'Rue de Zurich 43' avec un score de 81.25


2026-07-08 11:32:57.416 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue des Vieux Grenadiers 17' (seuil 95) : meilleure proposition 'Rue des Vieux-Grenadiers 17' avec un score de 66.67


2026-07-08 11:32:57.416 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue du 31 Décembre 68' (seuil 95) : meilleure proposition 'Rue du 31-Décembre 68' avec un score de 82.93


2026-07-08 11:32:57.417 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue du Fort Bareau, 19' (seuil 95) : meilleure proposition 'Rue du Fort-Barreau 19' avec un score de 61.9


2026-07-08 11:32:57.417 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue du Prieuré 15-17' (seuil 95) : meilleure proposition 'Rue du Prieuré 15' avec un score de 78.57


2026-07-08 11:32:57.417 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue du Village Suisse 3' (seuil 95) : meilleure proposition 'Rue du Village-Suisse 3' avec un score de 62.22


2026-07-08 11:32:57.418 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Vieux chemin d'Onex, 54' (seuil 95) : meilleure proposition 'Vieux-Chemin-d'Onex 54' avec un score de 64.0


2026-07-08 11:32:57.418 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'place Isaac Mercier, 1' (seuil 95) : meilleure proposition 'Place Isaac-MERCIER 1' avec un score de 85.11


2026-07-08 11:32:57.421 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:261 - 10/107 adresse(s) sans résultat SITG (voir les avertissements ci-dessus)


2026-07-08 11:32:57.422 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:267 - 38/107 adresse(s) trouvée(s) mais exclue(s) (score < 95)


2026-07-08 11:32:57.422 | INFO     | sitg_geocode.geocode:sitg_geocode_async:276 - 59/107 adresse(s) conservée(s) dans le résultat final


,Rue et N°,SITG_ADRESSE_ID,SITG_ADRESSE,SITG_NPA,SITG_NOM_NPA,SITG_COMMUNE,SITG_TYPE,SITG_CANTON,SITG_PAYS,SITG_EGID,SITG_EGRID,SITG_SCORE,SITG_PROVIDER,SITG_LON,SITG_LAT,SITG_EST_EPSG_2056,SITG_NORD_EPSG_2056
0,12 rue Jean-Charles Amat,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,"37, sous Garan",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,"Ancienne route, 78",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Av de Thonex 30,130404133145,Avenue de Thônex 30,1225.0,Chêne-Bourg,Chêne-Bourg,Existante,Canton de Genève,Suisse,295511715.0,CH576391476529,100.0,Système d'Information du Territoire à Genève (...,6.198379,46.190870,2504250.64,1116248.27
4,"Av. Blanc, 5",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
102,Ruelle du Midi 5,930225018798,Ruelle du Midi 5,1207.0,Genève,Genève-Eaux-Vives,Existante,Canton de Genève,Suisse,2039144.0,CH483965638718,100.0,Système d'Information du Territoire à Genève (...,6.157529,46.200510,2501114.79,1117370.48
103,"Vieux chemin d'Onex, 54",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
104,"place Isaac Mercier, 1",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
105,"rue Butini, 7",930225003901,Rue BUTINI 7,1202.0,Genève,Genève-Petit-Saconnex,Existante,Canton de Genève,Suisse,2037699.0,CH346365839449,100.0,Système d'Information du Territoire à Genève (...,6.149145,46.215521,2500495.16,1119049.65


## Utiliser geocoder-service (self-hosted) au lieu de l'API SITG Lab

[geocoder-service](https://github.com/denisiglesiasgarcia/geocoder-service) est un géocodeur auto-hébergé (Meilisearch + données ouvertes SITG `CAD_ADRESSE`), scopé au canton de Genève. Son endpoint `/api/v2/search` reproduit le même format de réponse que l'API SITG Lab, donc **aucun changement de code** n'est nécessaire côté `sitg-geocode` — seul le paramètre `api_url` change (ou la variable d'environnement `SITG_GEOCODE_API_URL`).

Prérequis : la stack `geocoder-service` démarrée localement (`docker compose --env-file .env up -d` dans son dépôt), exposée par défaut sur `http://localhost:8000`.

Différences à garder en tête par rapport à l'API SITG Lab :
- Scope Genève uniquement : pas de résultat hors canton (donc `canton=None` n'a pas d'effet particulier ici, tous les résultats sont déjà "Canton de Genève").
- `SITG_EGRID` est toujours `null` (pas de RF dans les données source, seulement RegBL/CAD_ADRESSE).
- Le score est calculé différemment (algorithme propre à `geocoder-service`, voir son README) : ne pas réutiliser tel quel un `min_score_threshold` calibré pour l'API SITG Lab.

In [3]:
from sitg_geocode import inspect_sitg_response

GEOCODER_SERVICE_URL = "http://localhost:8000/api/v2/search"

await inspect_sitg_response(adresse="Av. J-D Maillard, 7, Meyrin", api_url=GEOCODER_SERVICE_URL)

{
  "hits": [
    {
      "addressId": "930225017479",
      "streetName": "Avenue Jacob-Daniel-MAILLARD",
      "houseNumber": "7",
      "postalCode": "1217",
      "locality": "Meyrin",
      "municipality": "Meyrin",
      "type": "Existante",
      "EGID": "1021280",
      "EGRID": null,
      "longitude": 6.068229,
      "latitude": 46.22923,
      "coordinates": {
        "x": 2494278.54,
        "y": 1120679.36,
        "crs": "EPSG:2056"
      },
      "administrativeDivision": "Canton de Genève",
      "country": "Suisse",
      "score": 100.0,
      "dataSource": {
        "provider": "geocoder-service (SITG CAD_ADRESSE, données ouvertes)",
        "metadataUrl": "https://sitg.ge.ch/donnees/cad-adresse",
        "dataUrl": "https://ge.ch/sitg/geodata/SITG/OPENDATA/CAD_ADRESSE-SHP.zip"
      },
      "formatted": null
    }
  ],
  "query": "Av. J-D Maillard, 7, Meyrin",
  "processingTimeMs": 10.15,
  "limit": 1,
  "offset": 0,
  "nbHits": 1000
}


In [4]:
import polars as pl
from polars.dataframe.frame import DataFrame

from sitg_geocode import sitg_geocode_async

# même CSV de test que ci-dessus, mais géocodé via geocoder-service
df: DataFrame = pl.read_csv(source="test_adresses.csv")

result_geocode_geocoder_service: DataFrame = await sitg_geocode_async(
    df,
    col_adresse="Rue et N°",
    max_concurrent=10,
    min_score_threshold=95,
    api_url=GEOCODER_SERVICE_URL,
)

df_resultat_geocoder_service: DataFrame = df.join(
    other=result_geocode_geocoder_service, on="Rue et N°", how="left"
)

display(df_resultat_geocoder_service.to_pandas())

Geocoding SITG:   0%|          | 0/107 [00:00<?, ?it/s]

2026-07-08 11:32:59.507 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour '37, sous Garan' (seuil 95) : meilleure proposition 'Chemin de Sous-Caran 37' avec un score de 75.0


2026-07-08 11:32:59.508 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Av. Blanc, 5' (seuil 95) : meilleure proposition 'Avenue BLANC 53' avec un score de 71.55


2026-07-08 11:32:59.509 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Bvd Jaous Fazy 23' (seuil 95) : meilleure proposition 'Boulevard des Philosophes 23' avec un score de 49.44


2026-07-08 11:32:59.510 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin du Pont de l'Ecu 1' (seuil 95) : meilleure proposition 'Chemin du Pont-de-Ville 17' avec un score de 57.35


2026-07-08 11:32:59.510 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Chemin sous Blame, 53' (seuil 95) : meilleure proposition 'Chemin de Sous-Balme 53' avec un score de 93.37


2026-07-08 11:32:59.511 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Place Cornavin 1D' (seuil 95) : meilleure proposition 'Place de Cornavin 1' avec un score de 69.44


2026-07-08 11:32:59.511 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Route du Roussillon 18' (seuil 95) : meilleure proposition 'Route du Vallon 19' avec un score de 50.0


2026-07-08 11:32:59.511 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rte de grand Lancy 161' (seuil 95) : meilleure proposition 'Route de La-Chapelle 23' avec un score de 65.91


2026-07-08 11:32:59.512 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rte du Gd Lancy 166' (seuil 95) : meilleure proposition 'Route du Grand-Lancy 118' avec un score de 59.44


2026-07-08 11:32:59.512 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Jean-Jacque Rousseau 34' (seuil 95) : meilleure proposition 'Rue Jean-Jacques-De-SELLON 4' avec un score de 47.73


2026-07-08 11:32:59.513 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue Riont Parc 21' (seuil 95) : meilleure proposition 'Rue Robert-CÉARD 4' avec un score de 37.86


2026-07-08 11:32:59.513 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:246 - Aucune correspondance suffisante pour 'Rue du Simplon, 15' (seuil 95) : meilleure proposition 'Rue du Trabli 15' avec un score de 77.0


2026-07-08 11:32:59.515 | WARNING  | sitg_geocode.geocode:sitg_geocode_async:267 - 12/107 adresse(s) trouvée(s) mais exclue(s) (score < 95)


2026-07-08 11:32:59.516 | INFO     | sitg_geocode.geocode:sitg_geocode_async:276 - 95/107 adresse(s) conservée(s) dans le résultat final


,Rue et N°,SITG_ADRESSE_ID,SITG_ADRESSE,SITG_NPA,SITG_NOM_NPA,SITG_COMMUNE,SITG_TYPE,SITG_CANTON,SITG_PAYS,SITG_EGID,SITG_EGRID,SITG_SCORE,SITG_PROVIDER,SITG_LON,SITG_LAT,SITG_EST_EPSG_2056,SITG_NORD_EPSG_2056
0,12 rue Jean-Charles Amat,930225000455,Rue Jean-Charles-AMAT 12,1202.0,Genève,Genève-Petit-Saconnex,Existante,Canton de Genève,Suisse,2037721.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.148414,46.214420,2500436.72,1118928.13
1,"37, sous Garan",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,"Ancienne route, 78",930225000674,L'Ancienne-Route 78,1218.0,Le Grand-Saconnex,Grand-Saconnex,Existante,Canton de Genève,Suisse,2040113.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.124054,46.239172,2498603.30,1121710.62
3,Av de Thonex 30,130404133145,Avenue de Thônex 30,1225.0,Chêne-Bourg,Chêne-Bourg,Existante,Canton de Genève,Suisse,295511715.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.198379,46.190870,2504250.64,1116248.27
4,"Av. Blanc, 5",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
102,Ruelle du Midi 5,930225018798,Ruelle du Midi 5,1207.0,Genève,Genève-Eaux-Vives,Existante,Canton de Genève,Suisse,2039144.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.157529,46.200510,2501114.79,1117370.48
103,"Vieux chemin d'Onex, 54",930225031162,Vieux-Chemin-d'Onex 54,1213.0,Onex,Onex,Existante,Canton de Genève,Suisse,1022167.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.105423,46.185046,2497064.00,1115718.82
104,"place Isaac Mercier, 1",930225018382,Place Isaac-MERCIER 1,1201.0,Genève,Genève-Cité,Existante,Canton de Genève,Suisse,1010776.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.140075,46.206483,2499778.69,1118056.64
105,"rue Butini, 7",930225003901,Rue BUTINI 7,1202.0,Genève,Genève-Petit-Saconnex,Existante,Canton de Genève,Suisse,2037699.0,NaN,100.00,"geocoder-service (SITG CAD_ADRESSE, données ou...",6.149145,46.215521,2500495.16,1119049.65


## Comparaison SITG Lab vs. geocoder-service

Rapproche `df_resultat` (SITG Lab) et `df_resultat_geocoder_service` (geocoder-service) sur les mêmes 107 adresses de test, pour voir où les deux sont d'accord, où seul l'un des deux trouve un résultat, et où ils divergent sur l'adresse retenue.

In [5]:
comparison: DataFrame = (
    df.select("Rue et N°")
    .join(
        df_resultat.select("Rue et N°", "SITG_ADRESSE", "SITG_SCORE").rename(
            {"SITG_ADRESSE": "SITG_LAB_ADRESSE", "SITG_SCORE": "SITG_LAB_SCORE"}
        ),
        on="Rue et N°",
        how="left",
    )
    .join(
        df_resultat_geocoder_service.select("Rue et N°", "SITG_ADRESSE", "SITG_SCORE").rename(
            {"SITG_ADRESSE": "GEOCODER_SERVICE_ADRESSE", "SITG_SCORE": "GEOCODER_SERVICE_SCORE"}
        ),
        on="Rue et N°",
        how="left",
    )
)

n_total = comparison.height
n_sitg_lab_found = comparison["SITG_LAB_SCORE"].is_not_null().sum()
n_geocoder_service_found = comparison["GEOCODER_SERVICE_SCORE"].is_not_null().sum()
n_agree_adresse = comparison.filter(
    pl.col("SITG_LAB_ADRESSE") == pl.col("GEOCODER_SERVICE_ADRESSE")
).height

print(f"SITG Lab         : {n_sitg_lab_found}/{n_total} adresse(s) trouvée(s)")
print(f"geocoder-service  : {n_geocoder_service_found}/{n_total} adresse(s) trouvée(s)")
print(f"Même adresse retenue par les deux : {n_agree_adresse}/{n_total}")

# Cas où les deux services proposent une adresse différente : les plus intéressants à examiner.
disagreements = comparison.filter(
    pl.col("SITG_LAB_ADRESSE").is_not_null()
    & pl.col("GEOCODER_SERVICE_ADRESSE").is_not_null()
    & (pl.col("SITG_LAB_ADRESSE") != pl.col("GEOCODER_SERVICE_ADRESSE"))
)
n_disagreements = disagreements.height
print(f"\nDésaccord (résultat trouvé des deux côtés) : {n_disagreements}/{n_total}")
display(disagreements.to_pandas())

# Cas où un seul des deux services trouve un résultat.
only_sitg_lab = comparison.filter(
    pl.col("SITG_LAB_ADRESSE").is_not_null() & pl.col("GEOCODER_SERVICE_ADRESSE").is_null()
)
only_geocoder_service = comparison.filter(
    pl.col("SITG_LAB_ADRESSE").is_null() & pl.col("GEOCODER_SERVICE_ADRESSE").is_not_null()
)
print(f"\nTrouvé seulement par SITG Lab : {only_sitg_lab.height}/{n_total}")
display(only_sitg_lab.to_pandas())
print(f"\nTrouvé seulement par geocoder-service : {only_geocoder_service.height}/{n_total}")
display(only_geocoder_service.to_pandas())

SITG Lab         : 59/107 adresse(s) trouvée(s)
geocoder-service  : 95/107 adresse(s) trouvée(s)
Même adresse retenue par les deux : 57/107

Désaccord (résultat trouvé des deux côtés) : 2/107


,Rue et N°,SITG_LAB_ADRESSE,SITG_LAB_SCORE,GEOCODER_SERVICE_ADRESSE,GEOCODER_SERVICE_SCORE
0,Chemin des Crêts-de-Pregny,Chemin des Crêts-de-Pregny 16,100.0,Chemin des Crêts-de-Pregny 18,100.0
1,Rte de Sauverny,Route de Sauverny 223,100.0,Route de Sauverny 43,100.0



Trouvé seulement par SITG Lab : 0/107


,Rue et N°,SITG_LAB_ADRESSE,SITG_LAB_SCORE,GEOCODER_SERVICE_ADRESSE,GEOCODER_SERVICE_SCORE



Trouvé seulement par geocoder-service : 36/107


,Rue et N°,SITG_LAB_ADRESSE,SITG_LAB_SCORE,GEOCODER_SERVICE_ADRESSE,GEOCODER_SERVICE_SCORE
0,12 rue Jean-Charles Amat,NaN,NaN,Rue Jean-Charles-AMAT 12,100.00
1,"Ancienne route, 78",NaN,NaN,L'Ancienne-Route 78,100.00
2,"Av. J-D Maillard, 7",NaN,NaN,Avenue Jacob-Daniel-MAILLARD 7,100.00
3,Avenue de Châtelaine 60-62,NaN,NaN,Avenue de Châtelaine 60,100.00
4,Avenue du Grand Saleve 2,NaN,NaN,Avenue du Grand-Salève 2,100.00
5,"Ch de la Bassonette, 9",NaN,NaN,Chemin de la Bessonnette 9,100.00
6,Ch.Gustave Rochette 14,NaN,NaN,Chemin Gustave-ROCHETTE 14,100.00
7,Chemin Etienne Chennaz 10,NaN,NaN,Chemin Etienne-CHENNAZ 10,100.00
8,Chemin de la Montagne 51-53,NaN,NaN,Chemin De-LA-MONTAGNE 51,100.00
9,"Chemin de la Paumière,20",NaN,NaN,Chemin de la Paumière 20,100.00
